### PR TITLE
selectparser

### PR DESCRIPTION
**Pull request by Conorboyle8**
Added selectparser and added dataclass to statements. Also created test file for selectparser

### CODE REVIEW

File: src/selectparser.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from typing import List, Optional
+from statements import SelectStatement
+from baseparser import BaseParser, ParsingException
+from sqltoken import Token, TokenType
+
+
+class SelectParser(BaseParser):
+    def __init__(self, tokens: List[Token]):
```
Comment (samdojo) [inline src/selectparser.py:8]: [nit] `__init__()` function not needed but harmless

File: src/selectparser.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        # Expect SELECT keyword
+        if not (self.typeMatches(TokenType.KEYWORD) and self.valueMatches("SELECT")):
+            raise ParsingException("Expected SELECT")
+        self.consume(TokenType.KEYWORD, "SELECT")
+
+        # Parse columns
+        columns: List[str] = []
+        while self.tokens and not (self.typeMatches(TokenType.KEYWORD) and self.valueMatches("FROM")):
+            token = self.tokens.pop(0)
```
Comment (samdojo) [inline src/selectparser.py:23]: prefer `consume()`

File: src/test-selectparser.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        parser = SelectParser(tokens)
+        with self.assertRaises(Exception):
+            parser.parse()
+
+    def test_missing_columns(self):
+        tokens = tokenize("SELECT FROM users")  # missing columns
+        parser = SelectParser(tokens)
+        with self.assertRaises(Exception):
+            parser.parse()
+
```
Comment (samdojo) [inline src/test-selectparser.py:63]: I think you missed a couple of edge cases. Try `SELECT , FROM users`, `SELECT , id FROM users`, `SELECT id name FROM users`, etc.

File: src/selectparser.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        if not columns:
+            raise ParsingException("No columns specified in SELECT")
+
+        # Expect FROM keyword
+        if not (self.typeMatches(TokenType.KEYWORD) and self.valueMatches("FROM")):
+            raise ParsingException("Expected FROM")
+        self.consume(TokenType.KEYWORD, "FROM")
+
+        # Parse table name
+        if not self.tokens or not self.typeMatches(TokenType.IDENTIFIER):
```
Comment (samdojo) [inline src/selectparser.py:36]: need to use `TableOrSubQueryParser` here

File: src/selectparser.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            raise ParsingException("Expected table name")
+        table = self.consume(TokenType.IDENTIFIER).value.strip()
+
+        # Optional WHERE clause
+        where_clause: Optional[str] = None
+        if self.tokens and self.typeMatches(TokenType.KEYWORD) and self.valueMatches("WHERE"):
+            self.consume(TokenType.KEYWORD, "WHERE")
+
+            # Collect remaining token values as WHERE expression
+            where_tokens = []
```
Comment (samdojo) [inline src/selectparser.py:46]: need to use `ExpressionParser`

File: src/statements.py [hunk 6]
```diff
@@ -36,7 +36,9 @@ class DropTableStatement:
 
 @dataclass
 class SelectStatement:
-    pass
+    columns: List[str]
+    table: str
+    where: Optional[str] = None
```
Comment (samdojo) [inline src/statements.py:41]:
```suggestion
    where: List[Expression] = []
```

File: src/statements.py [hunk 7]
```diff
@@ -36,7 +36,9 @@ class DropTableStatement:
 
 @dataclass
 class SelectStatement:
-    pass
+    columns: List[str]
+    table: str
```
Comment (samdojo) [inline src/statements.py:40]:
```suggestion
    from: List[QualifiedTableName | SubQuery] = []
```

**Review by samdojo — CHANGES_REQUESTED** [review]

**Comment by samdojo** [comment]
Also—be sure to add one of your colleagues to this PR as well

**Review by samdojo — APPROVED** [review]
